src/prefect/server/logs/stream.py

```diff
@@ -0,0 +1,271 @@
+
+    # Check flow_run_id filter
+    if filter.flow_run_id:
+        if filter.flow_run_id.any_ is not None:
+            if log.flow_run_id not in filter.flow_run_id.any_:
+                return False
+
+    # Check task_run_id filter
+    if filter.task_run_id:
+        if filter.task_run_id.any_ is not None:
+            if log.task_run_id not in filter.task_run_id.any_:
+                return False
+        if filter.task_run_id.is_null_ is not None:
+            is_null = log.task_run_id is None
+            if filter.task_run_id.is_null_ != is_null:
+                return False
+
+    return True
+
+
+@asynccontextmanager
+async def distributor() -> AsyncGenerator[messaging.MessageHandler, None]:
+    """
+    Create a message handler that distributes logs to subscribed clients.
+
+    Yields:
+        A message handler function
+    """
+
+    async def message_handler(message: messaging.Message):
+        assert message.data
+
+        try:
+            assert message.attributes
+        except Exception:
+            return
+
+        if subscribers:
+            try:
+                log = Log.model_validate_json(message.data)
+            except Exception as e:
+                logger.warning(f"Failed to parse log message: {e}")
+                return
+
+            for queue in subscribers:
+                filter = filters[queue]
+                if not log_matches_filter(log, filter):
+                    continue
+
+                try:
+                    queue.put_nowait(log)
+                except asyncio.QueueFull:
+                    continue
+
+    yield message_handler
+
+
+_distributor_task: asyncio.Task[None] | None = None
+_distributor_started: asyncio.Event | None = None
+
+
+async def start_distributor() -> None:
+    """Starts the distributor consumer as a global background task"""
+    global _distributor_task
+    global _distributor_started
+    if _distributor_task:
+        return
+
+    _distributor_started = asyncio.Event()
+    _distributor_task = asyncio.create_task(run_distributor(_distributor_started))
+    await _distributor_started.wait()
+
+
+async def stop_distributor() -> None:
+    """Stops the distributor consumer global background task"""
+    global _distributor_task
+    global _distributor_started
+    if not _distributor_task:
+        return
+
+    task = _distributor_task
+    _distributor_task = None
+    _distributor_started = None
+
+    task.cancel()
+    try:
+        await asyncio.shield(task)
+    except asyncio.CancelledError:
+        pass
+
+
+class LogDistributor(RunInAllServers, Service):
+    """Service for distributing logs to websocket subscribers"""
+
+    name: str = "LogDistributor"
+
+    @classmethod
+    def service_settings(cls) -> ServicesBaseSetting:
+        raise NotImplementedError("LogDistributor does not have settings")
+
+    @classmethod
+    def environment_variable_name(cls) -> str:
+        return "PREFECT_API_LOGS_STREAM_OUT_ENABLED"
```

Comment:
im not sure to what extent we've already violated this pattern but just thinking out loud

the `AliasChoices` on the settings are meant to allow the old namespacing like `PREFECT_API_*` but with _new_ settings we've generally not added `PREFECT_API_*` as an `AliasChoices` since new settings would have never existed under the `API_` namespace, but its not an issue that they're accepted that

all of that to say, maybe the hardcoded variable name should use the new `SERVER_` namespace?

Reply:
Yes!